selfdrive/car/subaru/carcontroller.py

```diff
@@ -70,20 +130,42 @@ def update(self, CC, CS, now_nanos):
         self.es_distance_cnt = CS.es_distance_msg["COUNTER"]
 
     else:
-      if pcm_cancel_cmd and (self.frame - self.last_cancel_frame) > 0.2:
-        bus = 1 if self.CP.carFingerprint in GLOBAL_GEN2 else 0
-        can_sends.append(subarucan.create_es_distance(self.packer, CS.es_distance_msg, bus, pcm_cancel_cmd))
-        self.last_cancel_frame = self.frame
-
       if self.es_dashstatus_cnt != CS.es_dashstatus_msg["COUNTER"]:
-        can_sends.append(subarucan.create_es_dashstatus(self.packer, CS.es_dashstatus_msg))
+        can_sends.append(subarucan.create_es_dashstatus(self.packer, CS.es_dashstatus_msg, CC.enabled, CC.longActive, hud_control.leadVisible))
         self.es_dashstatus_cnt = CS.es_dashstatus_msg["COUNTER"]
 
-      if self.es_lkas_cnt != CS.es_lkas_msg["COUNTER"]:
-        can_sends.append(subarucan.create_es_lkas(self.packer, CS.es_lkas_msg, CC.enabled, hud_control.visualAlert,
+      if self.es_lkas_state_cnt != CS.es_lkas_state_msg["COUNTER"]:
+        can_sends.append(subarucan.create_es_lkas_state(self.packer, CS.es_lkas_state_msg, CC.enabled, hud_control.visualAlert,
                                                   hud_control.leftLaneVisible, hud_control.rightLaneVisible,
                                                   hud_control.leftLaneDepart, hud_control.rightLaneDepart))
-        self.es_lkas_cnt = CS.es_lkas_msg["COUNTER"]
+        self.es_lkas_state_cnt = CS.es_lkas_state_msg["COUNTER"]
+
+      if self.CP.openpilotLongitudinalControl:
+        if self.es_status_cnt != CS.es_status_msg["COUNTER"]:
```

Comment:
i wonder if it's worth it to just define the message rate and send at the fixed frequency, so we don't need all these counter variables

Reply:
The initial impreza port was done by comma, you could ask internally if anyone remembers why it was done this way